src/mantaray-v1.ts

```diff
@@ -0,0 +1,738 @@
+      this.makeDirty()
+      this._isEdge = true
+
+      return
+    }
+
+    /// Case when there is an existing fork for the given (sub)path
+
+    const commonPath = common(fork.prefix, path)
+    /** restPath of the existing fork's path that differs from the new path */
+    const restPath = fork.prefix.slice(commonPath.length)
+    let newNode = fork.node
+
+    // create new parent node where the path starts to differ in the current node's prefix
+    // this parent node will have two children: the current node and the new node with the given path
+    if (restPath.length > 0) {
+      // create new node for the common path
+      newNode = new MantarayNode()
+
+      if (equalBytes(this._obfuscationKey, null32Bytes)) {
+        newNode.obfuscationKey = new Uint8Array(32) as Bytes<32>
+      } else {
+        if (!obfuscationKeyGenerator) {
+          throw new RandomBytesFnUndefined()
+        }
+        newNode.obfuscationKey = obfuscationKeyGenerator()
+      }
+      newNode.forks = {}
+      const newFork = new MantarayFork(restPath, fork.node)
+
+      if (fork.node.isContinuousNode) {
+        handleTrimmedContinuousFork(newFork)
+      }
+      newNode.forks[restPath[0]] = newFork // copy old parent node to its remaining path
+      newNode._isEdge = true
+    }
+
+    // NOTE: special case on edge split
+    // newNode will be the common path edge node
+    // newNode's prefix is a subset of the given `path`, here the desired fork will be added with the truncated path
+    newNode.addFork(path.slice(commonPath.length), attributes)
+    this.forks[path[0]] = new MantarayFork(commonPath, newNode)
+    this._isEdge = true
+
+    this.makeDirty()
+  }
+
+  /**
+   * Gives back a MantarayFork under the given path
+   *
+   * @param path valid path within the MantarayNode
+   * @returns MantarayFork with the last unique prefix and its node
+   * @throws error if there is no node under the given path
+   */
+  public getForkAtPath(path: Uint8Array): MantarayFork {
+    if (path.length === 0) throw EmptyPathError
+
+    if (!this.forks) throw Error(`Fork mapping is not defined in the manifest`)
+
+    const fork = this.forks[path[0]]
+
+    if (!fork) throw new NotFoundError(path)
+
+    const prefixIndex = findIndexOfArray(path, fork.prefix)
+
+    if (prefixIndex === -1) throw new NotFoundError(path, fork.prefix)
+
+    const rest = path.slice(fork.prefix.length)
+
+    if (rest.length === 0) return fork
+
+    return fork.node.getForkAtPath(rest)
+  }
+
+  /**
+   * Removes a path from the node
+   *
+   * @param path Uint8Array of the path of the node intended to remove
+   */
+  public removePath(path: Uint8Array): void {
+    if (path.length === 0) throw EmptyPathError
+
+    if (!this.forks) throw Error(`Fork mapping is not defined in the manifest`)
+
+    const fork = this.forks[path[0]]
+
+    if (!fork) throw new NotFoundError(path)
+
+    const prefixIndex = findIndexOfArray(path, fork.prefix)
```

Comment:
Although the suggestion is closer, it is not true, because the first parameter has to encapsulate the second one.
I updated the JSDoc of the function instead.

Reply:
on the 3rd thought I created a different function for this logic, because it wasn't the most efficient.
btw, I checked out the code and it gives back the index where the 2nd bytearray starts in the 1st bytearray, so I renamed back the function.